.github/workflows/auto_update_charm_libs.yaml

```diff
@@ -24,31 +24,14 @@ jobs:
           charmcraft fetch-lib
         env:
           CHARMCRAFT_AUTH: ${{ secrets.CHARMHUB_TOKEN }}
-      - name: Import GPG key
-        id: import-gpg
-        uses: crazy-max/ghaction-import-gpg@v5
+      - name: Create pull request
+        uses: canonical/create-pull-request@main
         with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          passphrase: ${{ secrets.PASSPHRASE }}
-          git_config_global: true
-          git_commit_gpgsign: true
-          git_user_signingkey: true
-      - name: Create a PR for local changes
-        uses: peter-evans/create-pull-request@v4.2.3
-        env:
-          GIT_AUTHOR_NAME: ${{ steps.import-gpg.outputs.name }}
-          GIT_AUTHOR_EMAIL: ${{ steps.import-gpg.outputs.email }}
-          GIT_COMMITTER_NAME: ${{ steps.import-gpg.outputs.name }}
-          GIT_COMMITTER_EMAIL: ${{ steps.import-gpg.outputs.email }}
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
           commit-message: "chore: update charm libraries"
-          committer: ${{ env.GIT_COMMITTER_NAME }} <${{ env.GIT_COMMITTER_EMAIL }}>
-          author: ${{ env.GIT_AUTHOR_NAME }} <${{ env.GIT_AUTHOR_EMAIL }}>
-          title: "Update charm libraries"
+          branch-name: "chore/auto-libs"
```

Comment:
Is it possible to change the action's behaviour to update the branch if it already exists? That will make using it simpler and we'll want changes to be stacked in the same PR if there are changes

Reply:
Gotcha, let me create an option for that.

Reply:
I have added an option `upsert` to the `create-pull-request` action. When `upsert` is true, the branch and pull request will be updated instead of failing.

Reply:
Thank you! Updated